### PR TITLE
improve tests for v20 ledgers externalizing out of order

### PR DESCRIPTION
**Pull request by marta-lokhova**
Resolves https://github.com/stellar/stellar-core/issues/4012

### CODE REVIEW

File: src/herder/test/HerderTests.cpp [hunk 1]
```diff
@@ -4055,6 +4055,41 @@ checkHerder(Application& app, HerderImpl& herder, Herder::State expectedState,
...
+        LedgerTxn ltx(app.getLedgerTxnRoot());
+        ConfigUpgradeSetFrameConstPtr configUpgradeSet;
+        ConfigUpgradeSet configUpgradeSetXdr;
+        auto& configEntry = configUpgradeSetXdr.updatedEntry.emplace_back();
+        configEntry.configSettingID(CONFIG_SETTING_CONTRACT_HISTORICAL_DATA_V0);
+        configEntry.contractHistoricalData().feeHistorical1KB = 1234;
+        configUpgradeSet = makeConfigUpgradeSet(ltx, configUpgradeSetXdr);
+
+        scheduledUpgrades.mConfigUpgradeSetKey = configUpgradeSet->getKey();
+        // C won't upgrade until it's on the right LCL
```
Comment (dmkozh) [inline src/herder/test/HerderTests.cpp:4087]: Seems like a leftover comment.

**Review by dmkozh — APPROVED** [review]
Sorry, this got lost in my review pile.

**Comment by dmkozh** [comment]
r+ 093f37b

**Comment by marta-lokhova** [comment]
merge conflict, fixed

**Comment by dmkozh** [comment]
r+ 2ada673

**Comment by dmkozh** [comment]
r+ 567c8f9

**Pull request merged**